src/main/kotlin/com/zepben/evolve/cim/iec61970/base/wires/SynchronousMachine.kt

```diff
@@ -0,0 +1,130 @@
+ * @property satDirectSyncX Direct-axes saturated synchronous reactance (xdsat); reciprocal of short-circuit ration, as a percentage.
+ *                          Used for short circuit data exchange, only for single fed short circuit on a generator. (4.3.4.2. in IEC 60909-0:2001).
+ * @property satDirectTransX Saturated Direct-axis transient reactance as a percentage.
+ *                           The attribute is primarily used for short circuit calculations according to ANSI.
+ * @property x0 Zero sequence reactance of the synchronous machine as a percentage.
+ * @property x2 Negative sequence reactance as a percentage.
+ * @property type Modes that this synchronous machine can operate in.
+ * @property operatingMode Current mode of operation.
+ */
+class SynchronousMachine @JvmOverloads constructor(mRID: String = "") : RotatingMachine(mRID) {
+
+    private var _reactiveCapabilityCurve: MutableList<ReactiveCapabilityCurve>? = null
+
+    var baseQ: Double? = null
+    var condenserP: Int? = null
+    var earthing: Boolean? = null
+    var earthingStarPointR: Double? = null
+    var earthingStarPointX: Double? = null
+    var ikk: Double? = null
+
+    var maxQ: Double? = null
+    var maxU: Int? = null
+    var minQ: Double? = null
+    var minU: Int? = null
+
+    var mu: Double? = null
+    var r: Double? = null
+    var r0: Double? = null
+    var r2: Double? = null
+
+    var satDirectSubtransX: Double? = null
+    var satDirectSyncX: Double? = null
+    var satDirectTransX: Double? = null
+
+    var x0: Double? = null
+    var x2: Double? = null
+
+    var type: SynchronousMachineKind = SynchronousMachineKind.UNKNOWN
+    var operatingMode: SynchronousMachineKind = SynchronousMachineKind.UNKNOWN
+
+    /**
+     * All available [ReactiveCapabilityCurve] for this synchronous machine.
+     * First entry is the default [ReactiveCapabilityCurve]
+     */
+    val curves: Collection<ReactiveCapabilityCurve> get() = _reactiveCapabilityCurve.asUnmodifiable()
+
+    /**
+     * Get the number of entries in the [ReactiveCapabilityCurve] collection.
+     */
+    fun numCurves(): Int = _reactiveCapabilityCurve?.size ?: 0
+
+    /**
+     * The individual [ReactiveCapabilityCurve] for this [SynchronousMachine]
+     *
+     * @param mRID the mRID of the required [ReactiveCapabilityCurve]
+     * @return The [ReactiveCapabilityCurve] with the specified [mRID] if it exists, otherwise null
+     */
+    fun getCurve(mRID: String): ReactiveCapabilityCurve? = _reactiveCapabilityCurve?.getByMRID(mRID)
+
+    /**
+     * Add a [ReactiveCapabilityCurve] for this [SynchronousMachine]
+     *
+     * @param rcc the [ReactiveCapabilityCurve] to be added from this [SynchronousMachine]
+     */
+    fun addCurve(rcc: ReactiveCapabilityCurve): SynchronousMachine {
+
+        _reactiveCapabilityCurve = _reactiveCapabilityCurve ?: mutableListOf()
+        _reactiveCapabilityCurve!!.add(rcc)
```

Comment:
todo: This should have been updated to use your new `or` extension. Please check for others in your new code.